specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apimworkspacecertificates.json

```diff
@@ -0,0 +1,405 @@
+    "version": "2023-09-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/workspaces/{workspaceId}/certificates": {
+      "get": {
+        "tags": [
+          "Certificate"
+        ],
+        "operationId": "WorkspaceCertificate_ListByWorkspace",
+        "description": "Lists a collection of all certificates in the specified workspace.",
+        "x-ms-examples": {
+          "ApiManagementListWorkspaceCertificates": {
+            "$ref": "./examples/ApiManagementListWorkspaceCertificates.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/WorkspaceIdParameter"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "|     Field     |     Usage     |     Supported operators     |     Supported functions     |</br>|-------------|-------------|-------------|-------------|</br>| name | filter | ge, le, eq, ne, gt, lt | substringof, contains, startswith, endswith |</br>| subject | filter | ge, le, eq, ne, gt, lt | substringof, contains, startswith, endswith |</br>| thumbprint | filter | ge, le, eq, ne, gt, lt | substringof, contains, startswith, endswith |</br>| expirationDate | filter | ge, le, eq, ne, gt, lt |     |</br>"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "name": "isKeyVaultRefreshFailed",
```

Comment:
>isKeyVaultRefreshFailed [](http://example.com/codeflow?start=21&length=23)

Recommend using string enum over boolean. E.g. refreshState: { FailedOnly, All }.

Reply:
I think this is a good improvement but we want to have the same contract as service level certificates (`/service/{serviceName}/certificates` == `/service/{serviceName}/workspaces/{workspaceId}/certificates`) If we want to move to enum instead of boolean we need to move both of the endpoints. I uderstand that it would be a breaking change so we do not want to do that now.